src/main/java/com/autotune/analyzer/deployment/AutotuneDeployment.java

```diff
@@ -528,10 +528,24 @@ private static AutotuneConfig getAutotuneConfig(String autotuneConfigResource, K
 					}
 				}
 
-				String tunableName = tunableJson.optString(AutotuneConfigConstants.NAME);
-				String tunableValueType = tunableJson.optString(AutotuneConfigConstants.VALUE_TYPE);
-				String upperBound = tunableJson.optString(AutotuneConfigConstants.UPPER_BOUND);
-				String lowerBound = tunableJson.optString(AutotuneConfigConstants.LOWER_BOUND);
+				String tunableName = tunableJson.optString(AnalyzerConstants.AutotuneConfigConstants.NAME);
+				String tunableValueType = tunableJson.optString(AnalyzerConstants.AutotuneConfigConstants.VALUE_TYPE);
+
+				String upperBound = "";
+				String lowerBound = "";
+				List<String> choices = new ArrayList<>();
+
+				// check the tunable type and if it's categorical then we need to add the list of the values else we'll take the upper, lower bound values
+				if (tunableValueType.equalsIgnoreCase("categorical")) {
+					JSONArray categoricalChoicesJson = tunableJson.getJSONArray(AnalyzerConstants.AutotuneConfigConstants.TUNABLE_CHOICES);
+					for (Object categoricalChoiceObject : categoricalChoicesJson) {
+						String categoricalChoice = (String) categoricalChoiceObject;
+						choices.add(categoricalChoice);
+					}
+				} else {
+					upperBound = tunableJson.optString(AnalyzerConstants.AutotuneConfigConstants.UPPER_BOUND);
+					lowerBound = tunableJson.optString(AnalyzerConstants.AutotuneConfigConstants.LOWER_BOUND);
+				}
 				// Read in step from the tunable, set it to '1' if not specified.
 				double step = tunableJson.optDouble(AutotuneConfigConstants.STEP, 1);
```

Comment:
Step will not be applicable for categorical choices, please update

Reply:
Updated as requested